lib/ansible/module_utils/vmware.py

```diff
@@ -234,11 +234,11 @@ def compile_folder_path_for_object(vobj):
     thisobj = vobj
     while hasattr(thisobj, 'parent'):
         thisobj = thisobj.parent
+        if thisobj.name == 'Datacenters':
```

Comment:
This is not doing exactly the same thing (maybe on purpose ?).

Originally it would remove 'Datacenters' if it was the first itm in paths. Now it seems you're removing every item that contains 'Datacenters'. Can you confirm this change of implementation is exactly what you want ?

Reply:
As far as i know, you correct in those assumptions. Just worried what will happen if there is actually a subfolder with name 'Datacenters' (for instance: Datacenters/DC01/vm/IT-loads/Datacenters/London/)